### ISSUE TITLE
GatsbyImage's "onLoad" prop does not always trigger when image is not cached

### ISSUE THREAD

**Issue by dryan-FF** [gatsbyjs/gatsby#36373]
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/gatsbyjs/gatsby/issues
- [X] This issue is not a question, feature request, RFC, or anything other than a bug report directly related to Gatsby. Please post those things in GitHub Discussions: https://github.com/gatsbyjs/gatsby/discussions

### Description

our team is seeing an issue with the `GatsbyImage` component's `onLoad` prop when making our production site using `gatsby build`. On our site, we have rows of images, each of which is rendered via `<GatsbyImage>` and is given an `onLoad` prop to trigger some functionality in the parent component. When users open the page for the first time, this `onLoad` event listener does not trigger. The correct behavior does show when users reload the page (or when the images are otherwise cached)

I've tested our same code using the native `<img>` element's `onload` attribute with no issues. We also do not see this issue when running our site using `gatsby develop`

The below gifs demonstrate this behavior, following the steps to re-produce. Helpful information is displayed in the "Images Loaded" object printed on the page, and in the console log

### With Browser Cache Enabled

Here, you'll see that on "first page load", the `onLoad` event is not triggered. However, it is loaded on subsequent loads because the browser has the image cached

![with cache enabled](https://user-images.githubusercontent.com/81592986/184196930-02018c70-30bc-423c-84a2-a7cb31212e2e.gif)

### With Browser Cache Disabled

Here, each page load gives a different result: 
- first page load does not trigger any image's `onLoad` event
- second page load triggers all three image's `onLoad` event
- third page load only triggers the first image's `onLoad` event

![with cache disabled](https://user-images.githubusercontent.com/81592986/184196949-a071aae2-f22f-41c1-9b50-ddadc8ecc5a5.gif)

### Reproduction Link

https://github.com/dryan-FF/gatsby-image-onload-issue

### Steps to Reproduce

1. clone repository
2. run `npm ci && npm run build && npm run serve`
3. disable caching in the browser. In Chrome, this can be done in the network tab
4. open `http://localhost:9000/test`. To be safe, do so in a "private"/"incognito" window

### Expected Result

- When page loads, the 3 images load in
- each image triggers the onLoad prop passed in to them which updates the "Images Loaded" object.
- End result should be `{ 0: true, 1: true, 2: true }`

### Actual Result

- When page loads, the 3 images correctly load in, _however_ they do not each trigger their own `onLoad` prop
- behavior is inconsistent across page loads
- this issue does not occur once image is cached

### Environment

```shell
System:
    OS: macOS 12.3
    CPU: (12) x64 Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 16.14.2 - ~/.nvm/versions/node/v16.14.2/bin/node
    Yarn: 1.22.10 - /usr/local/bin/yarn
    npm: 8.5.0 - ~/.nvm/versions/node/v16.14.2/bin/npm
  Browsers:
    Chrome: 104.0.5112.79
    Firefox: 103.0.2
    Safari: 15.4
  npmPackages:
    gatsby: ^4.20.0 => 4.20.0
    gatsby-plugin-image: ^2.20.0 => 2.20.0
    gatsby-plugin-postcss: ^5.20.0 => 5.20.0
    gatsby-plugin-sharp: ^4.20.0 => 4.20.0
    gatsby-source-filesystem: ^4.20.0 => 4.20.0
    gatsby-transformer-sharp: ^4.20.0 => 4.20.0
```

**Comment by tyhopp** [comment]
Hi @dryan-FF, we published the fix for this in `gatsby-plugin-image@2.22.0-next.1`. You can install this version if you're interested in trying it out now, otherwise it will land in the next stable release.

**Comment by dryan-FF** [comment]
excellent, thank you!
